processor/src/test/java/org/mapstruct/ap/test/targetthis/CustomerEntityMapper1.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.targetthis;
+
+import org.mapstruct.CollectionMappingStrategy;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.ReportingPolicy;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(
+        unmappedSourcePolicy = ReportingPolicy.IGNORE,
+        unmappedTargetPolicy = ReportingPolicy.IGNORE,
+        collectionMappingStrategy = CollectionMappingStrategy.ADDER_PREFERRED
+)
+public interface CustomerEntityMapper1 {
```

Comment:
What would be interesting to see are testing with:
1. existing targets (`@MappingTarget`)
2. reverse mappings `@InheritInverseConfiguration)

Reply:
I updated mapper in test to include @InheritInverseConfiguration and @MappingTarget (for update).

Reply:
It was nice to see that @InheritInverseConfiguration generated exactly same code as with Mapping annotation. Just had to add name after putting there methods for update, as compiler got confused which one to use.